File: src/main/java/me/mucloud/plugin/MK/JoinMessage/IConfiguration.java
package me.mucloud.plugin.MK.JoinMessage;

import java.io.File;
import java.util.List;

public interface IConfiguration {

    String getVersion();

    File getConfigFile();

    void setJoinMessage(String message);

    void setExitMessage(String message);

    String getJoinMessage();

    String getExitMessage();

    List<PermissionMessage> getPermissionMessageMap();

    PermissionMessage comparePriority(List<PermissionMessage> pml);

    String messageChange(String message);

}

File: src/main/kotlin/me/mucloud/plugins/MK/JoinMessage/IConfiguration.kt
package me.mucloud.plugins.MK.JoinMessage

import java.io.File

interface IConfiguration {

    fun getVersion(): String

    fun getConfigFile(): File

    fun setJoinMessage(message: String?)

    fun setExitMessage(message: String?)

    fun getJoinMessage(): String?

    fun getExitMessage(): String?

    fun getPermissionMessageMap(): MutableList<PermissionMessage>?

    fun comparePriority(pml: MutableList<PermissionMessage>): PermissionMessage

    fun messageChange(message: String): String

}